tests/test_check_files_checksums.py

```diff
@@ -175,3 +160,89 @@ def should_log_error_when_validation_fails(
                 call().save(url, Check.CHECKSUM, ValidationResult.FAILED, details=expected_details),
             ]
         )
+
+
+@patch("backend.check_files_checksums.utils.S3_CLIENT.get_object")
+@patch("backend.check_files_checksums.utils.ProcessingAssetsModel")
+@patch("backend.check_files_checksums.task.ValidationResultFactory")
+def should_save_staging_access_validation_results(
+    validation_results_factory_mock: MagicMock,
+    processing_assets_model_mock: MagicMock,
+    get_object_mock: MagicMock,
+) -> None:
+    expected_error = ClientError(
+        {"Error": {"Code": "TEST", "Message": "TEST"}}, operation_name="get_object"
+    )
+    get_object_mock.side_effect = expected_error
+
+    s3_url = any_s3_url()
+    dataset_id = any_dataset_id()
+    version_id = any_dataset_version_id()
+    hash_key = f"DATASET#{dataset_id}#VERSION#{version_id}"
+
+    array_index = "1"
+
+    # When
+    environ[ARRAY_INDEX_VARIABLE_NAME] = array_index
```

Comment:
This affects other tests, so we should probably use [`patch.dict`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch.dict) to set it.

Could you also have a look at the other tests which set `environ[…]`? There are a few.